src/scripts/main.js

```diff
@@ -1,3 +1,224 @@
 'use strict';
 
-// write your code here
+const gameSize = 4;
+const startBtn = document.querySelector('.button');
+const startMessage = document.querySelector('.message-start');
+const loseMessage = document.querySelector('.message-lose');
+const winMessage = document.querySelector('.message-win');
+const scoreField = document.querySelector('.game-score');
+let score = 0;
+let gameSquare = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+
+];
+
+startBtn.addEventListener('click', () => {
+  newGame();
+
+  if (startBtn.innerText === 'Start') {
+    startBtn.innerText = 'Restart';
+    startBtn.classList.replace('start', 'restart');
+  }
+
+  startMessage.classList.add('hidden');
+  winMessage.classList.add('hidden');
+  loseMessage.classList.add('hidden');
+});
+
+function generateNewNumber() {
+  const randomRow = Math.floor(Math.random() * gameSize);
+  const randomCell = Math.floor(Math.random() * gameSize);
+
+  if (gameSquare[randomRow][randomCell] === 0) {
+    gameSquare[randomRow][randomCell] = Math.random() > 0.1 ? 2 : 4;
+  } else {
+    generateNewNumber();
+  }
+}
+
+function fillGameField() {
+  const gameBoard = document.querySelector('.game-field');
+  const rows = [...gameBoard.rows];
+  const e = [];
+
+  for (const a of rows) {
+    e.push([...a.cells]);
+  }
+
+  for (let i = 0; i < gameSize; i++) {
+    for (let x = 0; x < gameSize; x++) {
+      const gameValue = e[i][x];
+      const arrayValue = gameSquare[i][x];
+
+      gameValue.innerText = arrayValue === 0 ? '' : arrayValue;
```

Comment:
If you need to check just for the value, then you can write it down simpler.... If you need only for zero, and an empty string or `undefined` can pass, then leave the solution that was

```suggestion
      gameValue.innerText = !arrayValue ? '' : arrayValue;
```